stockroom/external/importer/torchvision_importers.py

```diff
@@ -81,3 +86,327 @@ class FashionMnist(TorchvisionCommon):
+
+    def dtypes(self):
+        return self.sample_img.dtype, self.sample_label.dtype
+
+    @classmethod
+    def gen_splits(cls, root):
+        return [cls(root)]
+
+    def __iter__(self):
+        for img, label in self.dataset:
+            yield self._process_data(img, label)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def variability_status(self):
+        return True, False
+
+    @staticmethod
+    def _process_data(img, lbl):
+        img = np.array(img)
+        lbl = np.array(lbl)
+        return img, lbl
+
+    def _gen_metadata(self):
+        print("Parsing data...")
+        H = W = 0
+        self.sample_counter = defaultdict(int)
+        for img, lbl in self.dataset:
+            # find max height/width
+            h, w = img.size
+            if h > H:
+                H = h
+            if w > W:
+                W = w
+
+            # find num samples in each class
+            self.sample_counter[lbl] += 1
+        _, _, c = self.sample_img.shape
+        self.shape = ((W, H, c), self.sample_label.shape)
+
+    def _print_table(self):
+        table = Table(title="Sample Distribution")
+
+        # columns
+        table.add_column("Class Name")
+        table.add_column("Number of samples", justify="right")
+
+        # rows
+        for class_label in self.dataset.classes:
+            num_samples = self.sample_counter[self.dataset.class_to_idx[class_label]]
+            table.add_row(str(class_label), str(num_samples))
+
+        console = Console()
+        console.print(table)
+
+
+def setup_coco_file(root: Path):
+    """
+    This is a Utility function for the Coco dataset. This function checks for any files
+    downloaded and if not it downloads and sets it up for you.
+    """
+    urls = {
+        "train.zip": "http://images.cocodataset.org/zips/train2017.zip",
+        "val.zip": "http://images.cocodataset.org/zips/val2017.zip",
+        "test.zip": "http://images.cocodataset.org/zips/test2017.zip",
+        "annotations.zip": "http://images.cocodataset.org/annotations/annotations_trainval2017.zip",
+    }
+
+    all_files_present = True
+    for file in urls:
+        if not os.path.exists(root / file):
+            all_files_present = False
+            break
+
+    # if not present download
+    if not all_files_present:
+        print("COCO files not found. Downloading...")
+        utils.download(urls, root)
+        files = [os.path.join(root, file) for file in urls]
+        utils.unzip(files, root)
+
+
+class COCOCaptions(BaseImporter):
+    name = "coco_captions"
+
+    def __init__(self, root, ann_file, split):
+        self.dataset = datasets.CocoCaptions(root, ann_file)
+        self.split = split
+        self.sample_img, self.sample_captions = self._process_data(*self.dataset[0])
+
+    def column_names(self):
+        return f"{self.name}-{self.split}-image", f"{self.name}-{self.split}-captions"
+
+    def shapes(self):
+        return ((3, 640, 640), None)
```

Comment:
Remove redundant paranthesis
```suggestion
        return (3, 640, 640), None
```